packages/core/data-transfer/lib/providers/local-strapi-source-provider/assets.ts

```diff
@@ -0,0 +1,26 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import { Duplex } from 'stream';
+
+const IGNORED_FILES = ['.gitkeep'];
+
+/**
+ * Generate and consume assets streams in order to stream each file individually
+ */
+export const createAssetsStream = (strapi: Strapi.Strapi): Duplex => {
+  const assetsDirectory = path.join(strapi.dirs.static.public, 'uploads');
+
+  return Duplex.from(
+    (function* () {
+      const files = fs.readdirSync(assetsDirectory).filter((file) => !IGNORED_FILES.includes(file));
+
+      for (const file of files) {
+        const filePath = path.join(assetsDirectory, file);
+        const stats = fs.statSync(filePath);
+        const stream = fs.createReadStream(filePath);
+
+        yield { file, path: filePath, stats, stream };
+      }
+    })()
+  );
+};
```

Comment:
```suggestion
import * as path from 'path';
import { stat, readdir, open } from 'node:fs/promises';
import { Duplex } from 'stream';

const IGNORED_FILES = ['.gitkeep'];

/**
 * Generate and consume assets streams in order to stream each file individually
 */
export const createAssetsStream = (strapi: Strapi.Strapi): Duplex => {
  const assetsDirectory = path.join(strapi.dirs.static.public, 'uploads');

  return Duplex.from(
    (async function* () {
      const files = (await readdir(assetsDirectory)).filter(
        (file) => !IGNORED_FILES.includes(file)
      );

      for await (const file of files) {
        const filePath = path.join(assetsDirectory, file);
        const stats = await stat(filePath);
        const fh = await open(filePath);

        yield { file, path: filePath, stats, stream: fh.createReadStream() };
      }
    })()
  );
};
```

We should try to avoid the sync methods as much as possible since they're blocking. If we use async everywhere, we'll probably give the output stream more time to work while we wait.

Reply:
Done :heavy_check_mark: